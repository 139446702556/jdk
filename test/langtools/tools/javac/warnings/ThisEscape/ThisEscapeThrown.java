/*
 * @test /nodynamiccopyright/
 * @bug 8015831
 * @compile/ref=ThisEscapeThrown.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeThrown.java
 * @summary Verify 'this' escape detection from a thrown 'this'
 */

public class ThisEscapeThrown extends RuntimeException {

    public ThisEscapeThrown(Object obj) {
        if (obj == null)
            throw this;
    }
}
