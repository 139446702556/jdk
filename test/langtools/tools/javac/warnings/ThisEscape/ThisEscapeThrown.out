ThisEscapeThrown.java:12:13: compiler.warn.possible.this.escape
1 warning
